app/models/interactive_run_state.rb

```diff
@@ -184,4 +225,8 @@ def self.default_answer(conditions)
   def add_key_if_nil
     self.key = InteractiveRunState.generate_key if self.key.nil?
   end
+
+  def interactive_state_url(host)
+    host + Rails.application.routes.url_helpers.api_v1_show_interactive_run_state_path(key: self.key)
+  end
```

Comment:
A slightly better approach, that abstracts the construction of the url is to do it this way:
`Rails.application.routes.url_helpers.api_v1_show_interactive_run_state_url(key: self.key, protocol: protocol, host: host_with_port)`
You can see this approach used a few times in the portal for example: https://github.com/concord-consortium/rigse/blob/e1449b04ca831dc023f678e1cdd57ed3f1f5bf57/app/models/portal/clazz.rb#L299

Reply:
I've changed it. I had used the previous approach as that's what we use in other places in LARA (and requires one less argument to be passed around).